lib/Models/BoxDrawing.ts

```diff
@@ -948,11 +1003,25 @@ export default class BoxDrawing {
         Cartesian3.dot(normalWc, scene.camera.direction) >= 0;
     };
 
-    side.onMouseOver = onMouseOver;
-    side.onMouseOut = onMouseOut;
-    side.onPick = onPick;
-    side.onDrag = moveBoxOnDragSide;
-    side.onRelease = onRelease;
+    const isTopOrBottomSide = axis === Axis.Z;
+
+    // Call enabledFn only if movement is is allowed for this side, otherwise call disabledFn
+    const ifActionEnabled = (
+      enabledFn: (...args: any[]) => any,
+      disabledFn?: (...args: any[]) => any
+    ) => {
+      return (...args: any[]) => {
+        return this.disableVerticalMovement && isTopOrBottomSide
+          ? disabledFn?.apply(this, args)
+          : enabledFn.apply(this, args);
+      };
```

Comment:
A helper function that performs the `enabledFn`  action if vertical movement is enabled, otherwise it performs the `disabledFn` action.